src/Device.vala

```diff
@@ -102,9 +135,76 @@ public class Power.Device : Object {
         }
     }
 
+    public string cent_to_string () {
+        if (percentage < 20) {
+            return _("Critical");
+        }
+
+        if (percentage < 40) {
+            return _("Low");
+        }
+
+        if (percentage < 60) {
+            return _("Good");
+        }
+
+        if (percentage < 80) {
+            return _("High");
+        }
+
+        return _("Full");
+    }
+
     private void update_properties () {
+        battery_level = upower_device.battery_level;
+        model = upower_device.model; // Can sometimes update eg when phone is trusted
         percentage = upower_device.percentage;
         state = upower_device.state;
+
+        switch (device_type) {
+            case UPS:
+                icon_name = "uninterruptible-power-supply";
+                break;
+            case MOUSE:
+                icon_name = "input-mouse";
+                break;
+            case KEYBOARD:
+                icon_name =  "input-keyboard";
+                break;
+            case PDA:
+            case PHONE:
+                icon_name =  "phone";
+                break;
+            case MEDIA_PLAYER:
+                icon_name =  "multimedia-player";
+                break;
+            case TABLET:
+                icon_name = "input-touchpad";
+                break;
+            case PEN:
+                icon_name =  "input-tablet";
+                break;
+            case GAMING_INPUT:
+                icon_name =  "input-gaming";
+                break;
+            case COMPUTER:
+                if (model == "iPad") {
+                    icon_name = "computer-tablet";
```

Comment:
This is obviously kind of a dirty little hack. I can remove this. It doesn't seem like there's a good way to figure out if the attached device is a tablet or laptop